analysis/pf_dataset.py

```diff
@@ -33,102 +38,117 @@ def get_pf_clinical_events(pharmacy_first_ids):
 # - chronic sinusitis and immunosuppressed individuals for acute sinusitis
 def get_uncomplicated_uti_denominator(index_date, selected_events, pregnancy_codelist):
     urt_code = ["1090711000000102"]
-    count_urt_6m = count_past_events(index_date, selected_events, urt_code, 6)
-    count_urt_12m = count_past_events(index_date, selected_events, urt_code, 12)
+    count_urt_6m = count_past_events(index_date, clinical_events, urt_code, 6)
+    count_urt_12m = count_past_events(index_date, clinical_events, urt_code, 12)
 
     age = patients.age_on(index_date)
     pregnancy_status = check_pregnancy_status(
-        index_date, selected_events, pregnancy_codelist
+        index_date, clinical_events, pregnancy_codelist
     )
 
     return (
         (age >= 16)
         & (age <= 64)
-        & (patients.sex.is_in(["female"]) & pregnancy_status.is_null())
-        | (count_urt_6m < 2)
-        | (count_urt_12m < 3)
+        & (patients.sex.is_in(["female"]) & (pregnancy_status == False))
+        & (count_urt_6m < 2)
+        & (count_urt_12m < 3)
     )
 
 
 def get_shingles_denominator(index_date, selected_events, pregnancy_codelist):
     age = patients.age_on(index_date)
     pregnancy_status = check_pregnancy_status(
-        index_date, selected_events, pregnancy_codelist
+        index_date, clinical_events, pregnancy_codelist
     )
 
-    return (age >= 18) & pregnancy_status.is_null()
+    return (age >= 18) & (pregnancy_status == False)
 
 
 def get_impetigo_denominator(index_date, selected_events, pregnancy_codelist):
     impetigo_code = ["48277006"]
     count_impetigo_12m = count_past_events(
-        index_date, selected_events, impetigo_code, 12
+        index_date, clinical_events, impetigo_code, 12
     )
 
     age = patients.age_on(index_date)
     pregnancy_status = check_pregnancy_status(
-        index_date, selected_events, pregnancy_codelist
+        index_date, clinical_events, pregnancy_codelist
     )
 
-    return (
-        (age >= 1)
-        | (pregnancy_status.is_not_null() & (age >= 16))
-        | (count_impetigo_12m < 2)
+    return ((age >= 1) & (count_impetigo_12m < 2)) & (
+        (pregnancy_status == False) | ((pregnancy_status == True) & (age >= 16))
+    )
+
+
+def check_pregnancy_status_for_debug(index_date, selected_events, pregnancy_codelist):
```

Comment:
you could get this here: `from pf_variables_library import check_pregnancy_status`